### PR TITLE
add test robot to remove entry from dns

### PR DESCRIPTION
**Pull request by B-Naoufal**

### CODE REVIEW

File: api/test/robot_tests/web_on_chain/web_on_chain.robot [hunk 1]
```diff
@@ -46,14 +46,26 @@ POST /websiteUploader/upload
...
     ...    ${API_URL}/my/domains/${WALLET_NICKNAME}
     ...    expected_status=${STATUS_OK}
     Should Contain    ${response.text}    ${website_url}
+    
+POST /cmd/executeFunction To remove the dns entry
+    ${argument}=    keywords.String To Arg    ${website_url}
+    ${data}=    Create Dictionary
+    ...    nickname=${WALLET_NICKNAME}
+    ...    name=deleteEntryFromDNS
+    ...    at=AS12tsy5TtDyGsWggDvqDLagxZqkuTiTPqWhqmrx7KSQNJwQSbnBA
```
Comment (Thykof) [inline api/test/robot_tests/web_on_chain/web_on_chain.robot:60]: this variable should not be hardcoded here: define a variable at the beginning of the file

idealy, we should call endpoint /network and save as a global variable the DNS address

File: int/config/config_network.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ testnet:
     - https://test.massa.net/api/v2
 
 buildnet:
-  DNS: AS12quYbKSv91DZoz6jL9QEswpqhsYhwYv4mt9XFXC5rx7Kww3DEH
+  DNS: AS12tsy5TtDyGsWggDvqDLagxZqkuTiTPqWhqmrx7KSQNJwQSbnBA #AS12quYbKSv91DZoz6jL9QEswpqhsYhwYv4mt9XFXC5rx7Kww3DEH
```
Comment (Thykof) [inline int/config/config_network.yaml:7]: when everything is working properly, don't forget to remove the comment

**Review by Thykof — CHANGES_REQUESTED** [review]

File: api/test/robot_tests/web_on_chain/web_on_chain.robot [hunk 3]
```diff
@@ -46,14 +46,26 @@ POST /websiteUploader/upload
...
+    ${response}=    POST
+    ...    ${API_URL}/cmd/executeFunction
+    ...    json=${data}
+    ...    expected_status=${STATUS_OK}
+    
+GET /all/domains 
+    ${response}=    GET
+    ...    ${API_URL}/all/domains
+    ...    expected_status=${STATUS_OK}
+    Should Contain    ${response.text}    flappy
```
Comment (Thykof) [inline api/test/robot_tests/web_on_chain/web_on_chain.robot:71]: why did you move this test at the end?

**Review by Thykof — CHANGES_REQUESTED** [review]

File: api/test/robot_tests/web_on_chain/web_on_chain.robot (hunk 3 above)
Comment (B-Naoufal) [reply api/test/robot_tests/web_on_chain/web_on_chain.robot:78]: i moved this one GET /my/domains/{nickname} because it check if the ci successfully added the new dns entry , so you want to remove a dns entry  after this test GET /my/domains/{nickname} but the order of GET /all/domains 
 is irrelevant

File: api/test/robot_tests/web_on_chain/web_on_chain.robot [hunk 4]
```diff
@@ -46,14 +46,26 @@ POST /websiteUploader/upload
...
-    ${response}=    GET
-    ...    ${API_URL}/all/domains
-    ...    expected_status=${STATUS_OK}
-    Should Contain    ${response.text}    flappy
-
 GET /my/domains/{nickname}
     ${response}=    GET
     ...    ${API_URL}/my/domains/${WALLET_NICKNAME}
     ...    expected_status=${STATUS_OK}
     Should Contain    ${response.text}    ${website_url}
```
[the remark this answers is not in the document]
Comment (B-Naoufal) [reply api/test/robot_tests/web_on_chain/web_on_chain.robot:53]: maybe it's not the top priority for the moment ;)

[the remark this answers is not in the document]
Comment (Thykof) [reply api/test/robot_tests/web_on_chain/web_on_chain.robot:53]: no important

**Review by Thykof — APPROVED** [review]

File: api/test/robot_tests/web_on_chain/web_on_chain.robot [hunk 5]
```diff
@@ -46,14 +46,33 @@ POST /websiteUploader/upload
...
+Fetch Current DNS
+    ${response}=    GET    
+    ...    ${API_URL}/massa/node
+    ...    expected_status=${STATUS_OK}
+    ${current_dns}    Set Variable    ${response.json()['dns']}
+    Set Global Variable    ${DNS}    ${current_dns}
+        
+POST /cmd/executeFunction To remove the dns entry
+    ${argument}=    keywords.String To Arg    ${website_url}
+    ${data}=    Create Dictionary
```
Comment (peterjah) [inline api/test/robot_tests/web_on_chain/web_on_chain.robot:64]: `Create Dictionary` what is this used for ?

Comment (B-Naoufal) [reply api/test/robot_tests/web_on_chain/web_on_chain.robot:64]: we create the Dictionary which contains all inputs needed for the callSC   

    ...    name=deleteEntryFromDNS
    ...    at=${DNS}
    ...    args=${argument}`

Comment (Thykof) [reply api/test/robot_tests/web_on_chain/web_on_chain.robot:64]: this is how we create an associative array

File: api/test/robot_tests/web_on_chain/web_on_chain.robot [hunk 6]
```diff
@@ -46,14 +46,33 @@ POST /websiteUploader/upload
...
     ...    ${API_URL}/my/domains/${WALLET_NICKNAME}
     ...    expected_status=${STATUS_OK}
     Should Contain    ${response.text}    ${website_url}
+
+Fetch Current DNS
+    ${response}=    GET    
+    ...    ${API_URL}/massa/node
+    ...    expected_status=${STATUS_OK}
+    ${current_dns}    Set Variable    ${response.json()['dns']}
+    Set Global Variable    ${DNS}    ${current_dns}
```
Comment (thomas-senechal) [inline api/test/robot_tests/web_on_chain/web_on_chain.robot:60]: Should be in the keywords.resource for this test suite, and called in the SetupSuite

Comment (thomas-senechal) [reply api/test/robot_tests/web_on_chain/web_on_chain.robot:60]: Or moved into this function : https://github.com/massalabs/station/blob/1034-rf-remove-test-website-at-the-end-of-the-test-suite/api/test/robot_tests/web_on_chain/keywords.resource#L24

File: api/test/robot_tests/web_on_chain/web_on_chain.robot [hunk 7]
```diff
@@ -46,14 +46,33 @@ POST /websiteUploader/upload
...
-
 GET /my/domains/{nickname}
     ${response}=    GET
     ...    ${API_URL}/my/domains/${WALLET_NICKNAME}
     ...    expected_status=${STATUS_OK}
     Should Contain    ${response.text}    ${website_url}
+
+Fetch Current DNS
+    ${response}=    GET    
+    ...    ${API_URL}/massa/node
```
Comment (thomas-senechal) [inline api/test/robot_tests/web_on_chain/web_on_chain.robot:57]: I can't find robot tests of this endpoints, could you open an issue to do it ?

Comment (thomas-senechal) [reply api/test/robot_tests/web_on_chain/web_on_chain.robot:57]: Same for  `GET /network` and `POST /network/{network}`

**Review by thomas-senechal — CHANGES_REQUESTED** [review]

File: api/test/robot_tests/web_on_chain/web_on_chain.robot (hunk 7 above)
Comment (B-Naoufal) [reply api/test/robot_tests/web_on_chain/web_on_chain.robot]: https://github.com/massalabs/station/issues/1054
